File: hybrid_jumping_robot/scripts/Robot.py
#!/usr/bin/env python3
from IMU_robot import IMU
from Actuator_robot import Actuator
from Vel_robot import Vel
import time

class Robot:

    def __init__(self, name):
        self.name = name
        self.imu = IMU('someimuname')
        self.actuator = Actuator()
        self.velocity = Vel()
        self.imu.listener()

    def get_rpy(self):
        print(self.imu.get_rpy())

    def get_vel(self):
        pass

    def set_vel(self):
        pass

    def get_actuator_position(self):
        pass

    def set_actuator_position(self):
        pass


if __name__ == '__main__':
    r = Robot('test')
    time.sleep(1)
    while 1:
        r.get_rpy()

File: hybrid_robot/src/Robot.cpp
#include "Robot.hpp"

Robot::Robot(const std::string &name): name(name) {}

void Robot::set_front_vel(Float64 vel) {
  velocity.update_target(vel);
}

void Robot::set_actuator_position(Float64 pos) {
//  actuator.set_extension_position(pos);
}
